src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/components/empty_state/__tests__/__snapshots__/empty_state.test.js.snap

```diff
@@ -91,7 +113,11 @@ exports[`EmptyState should render normally 1`] = `
             onClick={[Function]}
             type="button"
           >
-            Check for new data
+            <FormattedMessage
+              defaultMessage="Check for new data"
+              id="kbn.management.indexPattern.create.emptyState.checkData.button"
```

Comment:
There is a naming convention for ids according to which id should start from namespace. Additionally tool, that checks keys duplication, uses namespace in its logic. So we can not rid of `kbn`.

I will update ids according to your comment.

Reply:
There is one case where the id does not end with a type of message.  It is occurs for divided messages to save the context and/or to save the message formatting.
More details you can find here https://github.com/elastic/kibana/pull/20620/files#diff-bd1fc4e943e2578ffd59d1fda669b4c0R20 .

For such cases we defined the following naming convention to help vendors to understand the context clearly :
1. the main message id has the type on the second-to-last position, thereby identifying a divided phrase, and the last segment named `detail`:

`“kbn.management.editIndexPattern.scripted.deprecationLang.label.detail”: "Convert you scripted fields to {link} to avoid any problems."`

2. The inner message id has the type on the second-to-last position and the name of the variable from the placeholder in the main message (in this case `link`) as the last segment:

`“kbn.management.editIndexPattern.scripted.deprecationLang.label.link”: "Painless"`

Will it be still easy to map to mental model of the code and the UI, when we combine the type with the previous message segment in such cases?:

`kbn.management.editIndexPattern.scripted.deprecationLangLabel.detail`
`kbn.management.editIndexPattern.scripted.deprecationLangLabel.link`

@cjcenizal, what are your thoughts on this?

Reply:
Thanks for the link! I should have read this sooner. This helps me understand the naming convention much better now.

Yes, I think that combining the type with the previous message segment still results in an easy-to-map mental model. What do you think of allowing people to add additional adjectives or descriptors to the last segment, if necessary? This could be useful for very complex blocks of text, or maybe for multiple paragraphs? For example:

`kbn.management.editIndexPattern.scripted.deprecationLangLabel.deprecationDetail`
`kbn.management.editIndexPattern.scripted.deprecationLangLabel.painlessLink`
`kbn.management.editIndexPattern.scripted.deprecationLangLabel.luceneLink`
`kbn.management.editIndexPattern.scripted.deprecationLangLabel.scriptedFieldsDetail`

Reply:
I updated message ids.

Concerning additional message types (such as `button`, `link`, `label` and so on) I think that it's up to a developer.  If one more type is required by particular case, it's ok.